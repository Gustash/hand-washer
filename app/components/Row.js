import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { colors } from 'app/config'

const Row = ({
  highlight,
  highlightStyle,
  primaryText,
  secondaryText,
  style,
  separator = false,
}) => (
  <View style={[styles.container, separator && styles.separator, style]}>
    <View
      testId="row-highlight-container"
      style={[styles.highlightContainer, highlightStyle]}
    >
      <Text testId="row-highlight-text" style={styles.highlightText}>
        {highlight}
      </Text>
    </View>
    <View style={styles.textContainer}>
      <Text testId="row-primary-text" style={styles.primaryText}>
        {primaryText}
      </Text>
      <Text testId="row-secondary-text" style={styles.secondaryText}>
        {secondaryText}
      </Text>
    </View>
  </View>
)

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 16,
  },
  separator: {
    borderBottomColor: colors.separator,
    borderBottomWidth: 1,
  },
  highlightContainer: {
    backgroundColor: colors.green,
    paddingHorizontal: 14,
    paddingVertical: 8,
    width: 80,
    justifyContent: 'center',
    marginHorizontal: 8,
  },
  highlightText: {
    textAlign: 'center',
    color: colors.white,
    fontSize: 17,
  },
  primaryText: {
    fontSize: 17,
    color: colors.darkGrey,
  },
  textContainer: {
    flex: 1,
  },
  secondaryText: {
    fontSize: 14,
    color: colors.lightGrey,
  },
})

export default Row
