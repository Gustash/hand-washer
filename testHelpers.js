export const findNodeWithTestId = (wrapper, testId) =>
  wrapper.findWhere(node => node.prop('testId') === testId)
