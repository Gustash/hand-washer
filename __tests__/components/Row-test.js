import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Row from '../../app/components/Row'
import { findNodeWithTestId } from '../../testHelpers'

const mockProps = props => ({
  highlight: 'Highlight',
  highlightStyle: { marginTop: 1000 },
  primaryText: 'Primary Text',
  secondaryText: 'Secondary Text',
  style: { marginBottom: 1000 },
  ...props,
})

describe('<Row />', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = mockProps()
    wrapper = shallow(<Row {...props} />)
  })

  it('matches snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('applies highlight text', () => {
    expect(
      findNodeWithTestId(wrapper, 'row-highlight-text').prop('children')
    ).toBe('Highlight')

    wrapper.setProps({
      ...props,
      highlight: 'Highlight 2',
    })

    expect(
      findNodeWithTestId(wrapper, 'row-highlight-text').prop('children')
    ).toBe('Highlight 2')
  })

  it('applies primary text', () => {
    expect(
      findNodeWithTestId(wrapper, 'row-primary-text').prop('children')
    ).toBe('Primary Text')

    wrapper.setProps({
      ...props,
      primaryText: 'Another Primary',
    })

    expect(
      findNodeWithTestId(wrapper, 'row-primary-text').prop('children')
    ).toBe('Another Primary')
  })

  it('applies secondary text', () => {
    expect(
      findNodeWithTestId(wrapper, 'row-secondary-text').prop('children')
    ).toBe('Secondary Text')

    wrapper.setProps({
      ...props,
      secondaryText: 'Another Secondary',
    })

    expect(
      findNodeWithTestId(wrapper, 'row-secondary-text').prop('children')
    ).toBe('Another Secondary')
  })

  it('applies highlight style', () => {
    expect(
      findNodeWithTestId(wrapper, 'row-highlight-container').prop('style')
    ).toContainEqual({
      marginTop: 1000,
    })

    wrapper.setProps({
      ...props,
      highlightStyle: {
        marginTop: 3000,
      },
    })

    expect(
      findNodeWithTestId(wrapper, 'row-highlight-container').prop('style')
    ).toContainEqual({
      marginTop: 3000,
    })
  })

  it('applies style', () => {
    expect(wrapper.first().prop('style')).toContainEqual({
      marginBottom: 1000,
    })

    wrapper.setProps({
      ...props,
      style: {
        marginBottom: 3000,
      },
    })

    expect(wrapper.first().prop('style')).toContainEqual({
      marginBottom: 3000,
    })
  })

  it('defaults separator to false', () => {
    expect(wrapper.first().prop('style')).toContain(false)
  })

  it('applies separator', () => {
    wrapper.setProps({
      ...props,
      separator: false,
    })

    expect(wrapper.first().prop('style')).not.toContainEqual({
      borderBottomColor: '#E6E6E6',
      borderBottomWidth: 1,
    })

    wrapper.setProps({
      ...props,
      separator: true,
    })

    expect(wrapper.first().prop('style')).toContainEqual({
      borderBottomColor: '#E6E6E6',
      borderBottomWidth: 1,
    })
  })
})
