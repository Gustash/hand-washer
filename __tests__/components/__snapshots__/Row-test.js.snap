// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "paddingVertical": 16,
      },
      false,
      Object {
        "marginBottom": 1000,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#00A778",
          "justifyContent": "center",
          "marginHorizontal": 8,
          "paddingHorizontal": 14,
          "paddingVertical": 8,
          "width": 80,
        },
        Object {
          "marginTop": 1000,
        },
      ]
    }
    testId="row-highlight-container"
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 17,
          "textAlign": "center",
        }
      }
      testId="row-highlight-text"
    >
      Highlight
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 17,
        }
      }
      testId="row-primary-text"
    >
      Primary Text
    </Text>
    <Text
      style={
        Object {
          "color": "#9B9B9B",
          "fontSize": 14,
        }
      }
      testId="row-secondary-text"
    >
      Secondary Text
    </Text>
  </View>
</View>
`;
